packages/electron/src/electron/main/index.ts

```diff
@@ -599,3 +593,55 @@ Promise.all([
 
   ipcMain.on("set-badge-count", handleBadgeCount);
 });
+
+const MAX_RETRIES = 3;
+const RECHECK_DELAY = 5000;
+const RETRY_DELAY_INCREMENT = 2000;
+
+async function makeSureSidecarIsRunning() {
+  let retries = 0;
+  let subscriber: NodeJS.Timeout | null = null;
+
+  async function handler() {
+    try {
+      /**
+       * Checking if the sidecar is running
+       * And if not, starting it up
```

Comment:
same as my comment below. if we want to add comments, comments should answer:

- Why did Aid add this? 
- Because the user could have a sidecar from another spotlight service already running